modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/upload/DDMFormUploadFileEntryHandler.java

```diff
@@ -0,0 +1,132 @@
+import com.liferay.dynamic.data.mapping.form.web.internal.constants.DDMFormConstants;
+import com.liferay.dynamic.data.mapping.form.web.internal.security.permission.resource.DDMFormInstancePermission;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Repository;
+import com.liferay.portal.kernel.portletfilerepository.PortletFileRepositoryUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.repository.model.Folder;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.upload.UploadPortletRequest;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.MimeTypesUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.upload.UploadFileEntryHandler;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Roberto Díaz
+ */
+@Component(service = DDMFormUploadFileEntryHandler.class)
+public class DDMFormUploadFileEntryHandler implements UploadFileEntryHandler {
+
+	@Override
+	public FileEntry upload(UploadPortletRequest uploadPortletRequest)
+		throws IOException, PortalException {
+
+		File file = null;
+
+		try (InputStream inputStream = uploadPortletRequest.getFileAsStream(
+				"file")) {
+
+			long formInstanceId = ParamUtil.getLong(
+				uploadPortletRequest, "formInstanceId");
+			long groupId = ParamUtil.getLong(uploadPortletRequest, "groupId");
+
+			String fileName = uploadPortletRequest.getFileName("file");
+
+			file = FileUtil.createTempFile(inputStream);
+
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)uploadPortletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			return addFileEntry(
+				formInstanceId, groupId, fileName,
+				MimeTypesUtil.getContentType(file, fileName), file,
+				themeDisplay);
+		}
+		finally {
+			FileUtil.delete(file);
+		}
+	}
+
+	protected FileEntry addFileEntry(
+			long formInstanceId, long groupId, String fileName, String mimeType,
+			File file, ThemeDisplay themeDisplay)
+		throws PortalException {
+
+		if (!DDMFormInstancePermission.contains(
+				themeDisplay.getPermissionChecker(), formInstanceId,
+				DDMActionKeys.ADD_FORM_INSTANCE_RECORD)) {
+
+			throw new PrincipalException.MustHavePermission(
+				themeDisplay.getPermissionChecker(),
+				DDMFormInstance.class.getName(), formInstanceId,
+				DDMActionKeys.ADD_FORM_INSTANCE_RECORD);
+		}
+
+		long userId = _userLocalService.getDefaultUserId(
```

Comment:
@carolmariaabb, just a note: on this line, we're getting the user ID which will be used to create the folder and the file entry. Maybe, we can change this code later to handle the issue solved by Paulino that created private folders to each user uploading files to Forms. How about get the ID of the logged user? Otherwise, we can get the ID of the user created by you, Forms Guest User.

Reply:
Hi @natocesarrego. I believe I understand what you mean. However, this code is only executed if the upload field is used by guest users. Otherwise, `DLUploadFileEntryHandler` is executed.